### DISCUSSION TITLE
Unable to localize content in liquid html part

### DISCUSSION THREAD

**Discussion by Notsoholy** [OrchardCMS/OrchardCore#15818] [Q&A] [answered]
Hello! I'm currently working on a project where we are using Orchard CMS for displaying content to our users. Currently we need to provide localized content based on Content-Language HTTP header, and we have a following structure:
We are currently using admin theme, and we have defined several content types and content items based on those types. Content type that needs localizing consists of several text fields and a liquid body part, which contains plain HTML. I've created a PO file, which was placed in Localization/ folder, "de.po", and added several lines like:
```
msgid "Help.Password"
msgtext "Passwort"
```
After this, in liquid body part I've replaced word "Passwort" with `{{ "Help.Password" | t }}`, however, in preview there were no changes, and I can only see my localization key. I can see in logs that it attempts to find translation, however, it fails, therefore it fallbacks to a key. I've tried to play around with msgctxt, however that didn't help. Also I've tried to add this text to `TheAdmin.po`, but it was overwritten during the build stage. Could you please advise on this topic? Thank you!

**Comment by Piedone** [comment]
@hishamco you can help here, perhaps?

**Comment by hishamco** [comment, marked as answer]
@Notsoholy did you enable the localization module, and then add your culture as a supported culture?

**Reply by Notsoholy** [reply]
Yes, I've enabled everything related to localization, I believe
![image](https://github.com/OrchardCMS/OrchardCore/assets/71740530/33c43b05-79f9-473e-9390-4dbef55fce5a)

**Reply by hishamco** [reply]
> Also I've tried to add this text to TheAdmin.po, but it was overwritten during the build stage

Yes it is

If you enable the localization feature and added the culture, please write a failing unit test or create a branch that I can test

**Reply by Notsoholy** [reply]
@hishamco @Piedone So translations are apparently working now, thank you! After debugging, I found out that during preview, culture was set to "en" by default, and therefore, there were no translations for my keys in English. I've copied my translations to `en.po` file, and everything is working now :) Now I just need to find out how to correctly set culture

**Reply by hishamco** [reply]
If you add your culture in the settings it should work

**Reply by Notsoholy** [reply]
I have successfully resolved this with sending `Accept-Language` header, thank you!
